src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -104,157 +112,167 @@ class GslQuadAdaptiveImpl {
  protected:
   size_t max_intervals_;
   gsl_function integrand_;
-
   std::unique_ptr<gsl_integration_workspace, gsl_integration_workspace_deleter>
       workspace_;
 };
 }  // namespace detail
 
+enum class IntegralType {
+  StandardGaussKronrod,            // gsl_integration_qag()
+  IntegrableSingularitiesPresent,  // gsl_integration_qags()
+  IntegrableSingularitiesKnown,    // gsl_integration_qagp()
+  InfiniteInterval,                // gsl_integration_qagi()
+  UpperBoundaryInfinite,           // gsl_integration_qagiu()
+  LowerBoundaryInfinite            // gsl_integration_qagil()
+};
+
 /*!
  * \ingroup NumericalAlgorithmsGroup
- * \brief A adaptive integration procedure class
+ * \brief A wrapper around the GSL adaptive integration procedures
```

Comment:
Does this general documentation and the individual template specializations show up properly in Doxygen? Please check

Reply:
I checked.  It works.

Reply:
I forgot to answer after checking. Sorry about that.